wp-parsely.php

```diff
@@ -47,12 +226,43 @@ function maybe_load_plugin() {
 	 *
 	 * add_filter( 'wpvip_parsely_load_mu', '__return_false' );
 	 */
-	if ( ! apply_filters( 'wpvip_parsely_load_mu', get_option( '_wpvip_parsely_mu' ) === '1' ) ) {
+
+	// The activation of this plugin has been queued in this request.
+	$self_managed_plugin_is_queued_for_activation = is_queued_for_activation();
+
+	// Check if wp-parsely has already been loaded, for example when the user
+	// has a self-managed wp-parsely installation.
+	$self_managed_plugin_has_loaded = in_array( PARSELY_PLUGIN_SIGNATURE, get_option( 'active_plugins', [] ) );
+
+	if ( $self_managed_plugin_has_loaded || $self_managed_plugin_is_queued_for_activation ) {
+		$parsely_options = Parsely_Loader_Info::get_parsely_options();
+		Parsely_Loader_Info::set_active( true );
```

Comment:
My only remaining question here is whether we should execute this for `$self_managed_plugin_is_queued_for_activation`, instead of just returning. @chriszarate, @mjangda, opinions?

Reply:
> My only remaining question here is whether we should execute this for `$self_managed_plugin_is_queued_for_activation`, instead of just returning. @chriszarate, @mjangda, opinions?

Adding @rinatkhaziev and @WPprodigy. They may have some thoughts on the question by @acicovic

Reply:
Thanks!

Reply:
All the `is_queued_for_activation()` related code paths (and tests) feels like it's adding more complication than is worth IMO. Can we just strip that part out? I don't _think_ we really need to know right that second on the activation requests themselves. The next PHP request will know about it's activated/deactivation, and we're never going to be activating & syncing sds data on the same request.

Reply:
This part was not added to record the info for SDS. `is_queued_for_activation()` was added because when a user activates a self managed Parse.ly plugin installation, and the code has `add_filter( 'wpvip_parsely_load_mu', '__return_true' );` somewhere in their code, there will be a class loading collision between the mu-plugins integrated wp-parsely, and their self-managed installation.
![Screen Shot 2022-08-26 at 9 21 12 AM](https://user-images.githubusercontent.com/46167019/188753159-f1016caa-dbd7-476f-b358-8c77fce63802.png)

Reply:
> ...and we're never going to be activating & syncing sds data on the same request.

@acicovic, I think this is the answer we've been seeking on your question. Based on @WPprodigy's input, there will not be a need to populate this data for use in the SDS request, because there will not be a sync to SDS on this request. I will make the necessary change that will short circuit this logic when activation is queued on this request.

Reply:
I have addressed this in my latest commit. [The SDS info will not be captured when the request has an activation queued](https://github.com/Automattic/vip-go-mu-plugins/pull/3542/files#diff-41f0c604e0848ecb9ac1b6e07d5d2fab90b2ef3f715792c15243c5eed180894cR249).

Reply:
Great, thank you both!